linera-service/src/cli_wrappers/docker.rs

```diff
@@ -12,32 +11,31 @@ pub struct DockerImage {
 }
 
 impl DockerImage {
-    pub async fn new(name: String, bin_path: &PathBuf, github_root: &PathBuf) -> Result<Self> {
-        let docker_image = Self { name };
-        docker_image.build(bin_path, github_root).await?;
-        Ok(docker_image)
-    }
-
     pub fn name(&self) -> &String {
         &self.name
     }
 
-    async fn build(&self, bin_path: &PathBuf, github_root: &PathBuf) -> Result<()> {
+    pub async fn build(name: String, bin_path: &PathBuf, github_root: &PathBuf) -> Result<Self> {
+        let docker_image = Self { name: name.clone() };
         let bin_path = diff_paths(bin_path, github_root).context("Getting relative path failed")?;
         let binaries_arg = format!(
             "binaries={}",
             bin_path.to_str().context("Getting str failed")?
         );
 
-        Command::new("docker")
+        let status = Command::new("docker")
             .current_dir(github_root)
             .arg("build")
             .args(["-f", "docker/Dockerfile"])
             .args(["--build-arg", &binaries_arg])
             .arg(".")
-            .args(["-t", &self.name])
-            .spawn_and_wait_for_stdout()
+            .args(["-t", &name])
+            .status()
             .await?;
-        Ok(())
+
+        if !status.success() {
+            return Err(anyhow::anyhow!("Error Docker building image {}", name,));
```

Comment:
Please use `anyhow::bail!` instead of `return Err(anyhow::anyhow!(...))`.

The test `status.success()` is repeated in many places. It seems like we could either keep using `spawn_and_wait_for_stdout` or create `spawn_and_wait` in `CommandExt` of `util.rs`.

Reply:
Good points, let me adjust